src/api/app.js

```diff
@@ -46,3 +65,45 @@ app.get('/', function (req, res) {
 
 const port = 3000;
 app.listen(port, () => logger.info(`Start service on port ${port}!`));
+
+function removeEmailIfOlderThan(email, dateTimeBoundary) {
+    return email["datetime"] > dateTimeBoundary;
+}
+
+function filterByEmail(email, to) {
+    let actualToEmail = email["personalizations"][0]["to"][0]["email"];
+    return actualToEmail === to;
+}
+
+function filterBySubject(email, subject) {
+    let actualSubject = email["subject"];
+    if(subject.startsWith('%') && subject.endsWith('%')) {
+        searchSubject = subject.substring(1, subject.length - 1);
+        console.log('Searching for emails with subject containing', searchSubject);
+        return actualSubject.includes(searchSubject);
+    }
+    return actualSubject === subject;
+}
+
+function filterByDateTimeSince(email, dateTimeSinceAsString) {
+    let dateTimeSince = Date.parse(dateTimeSinceAsString);
+    if(dateTimeSince === NaN) {
+        throw "The provided date cannot be parsed";
+    }
+    return email["datetime"] > dateTimeSince;
+}
+
+function parseDurationString( durationString ){
+    var stringPattern = /^PT(?:(\d+)D)?(?:(\d+)H)?(?:(\d+)M)?(?:(\d+(?:\.\d{1,3})?)S)?$/;
+    var stringParts = stringPattern.exec( durationString );
```

Comment:
Can you please trim the `durationString` before checking against the pattern:

```suggestion
    var stringParts = stringPattern.exec(durationString.trim());
```